src/main/java/seedu/address/model/person/DateOfLastVisit.java

```diff
@@ -3,32 +3,26 @@
 import static java.util.Objects.requireNonNull;
 import static seedu.address.commons.util.AppUtil.checkArgument;
 
-
+import java.time.DateTimeException;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.ResolverStyle;
 
 /**
  * Represents a Person's last visited date by the social worker in the address book.
  * Guarantees: immutable; is valid as declared in {@link #isValidDateOfLastVisit(String)}
  */
 public class DateOfLastVisit implements Comparable<DateOfLastVisit> {
 
-    public static final String MESSAGE_CONSTRAINTS = "Date of last visit should be in dd-MM-yyyy format.";
-
-    // THIS VALIDATION REGEX WAS GENERATED BY CHATGPT - ACKNOWLEDGE IN DG
-    // date must be in the format dd-MM-yyyy
-    public static final String VALIDATION_REGEX =
-            "^((0[1-9]|[12][0-9]|3[01])-(01|03|05|07|08|10|12)-\\d{4})|"
-                    + "((0[1-9]|[12][0-9]|30)-(04|06|09|11)-\\d{4})|"
-                    + "((0[1-9]|1[0-9]|2[0-8])-02-\\d{4})|"
-                    + "(29-02-(?:(?:[0-9]{2}(?:0[48]|[2468][048]|[13579][26]))|(?:[0-9]{2}00)))$";
-    // 31st day of months with 31 days
-    // 30th day of months with 30 days
-    // 01-29 for February (ignores leap year handling)
-    // Days 01-31 for other months
-    // -------------------
+    public static final String MESSAGE_CONSTRAINTS = "Date of last visit should be in dd-MM-yyyy format.\n"
+            + "Ensure that the month is 01-12 and the date is not later than today. ";
```

Comment:
I don't quite understand the ensure month is 01-12 comment. Do you mean ensure that month is after day?

Reply:
yup that was the intention and i will update the error message as suggested